### PR TITLE
adding v602 and s302 fixes for generating a clean file

### LINKED ISSUE #640: Fix validation of v602 and s302 for clean file generation.
String formatting would need to be made with an apply function.

### PR DESCRIPTION
**Pull request by MichaelFurey17**
Close #640

### CODE REVIEW

File: python/rules_engine.py [hunk 1]
```diff
@@ -190,7 +190,8 @@ def s302(self):
 		"""The reported Calendar Year does not match the filing year indicated at the start of the filing."""
 		field = "calendar_year"
 		edit_name = "s302"
-		fail_df = self.ts_df[self.ts_df.calendar_year != str(self.year)]
+		year = str(self.year)
```
Comment (Kibrael) [inline python/rules_engine.py]: Does this actually change how it functions? The logic here looks the same, curious if it's hiding something else?

File: python/rules_engine.py [hunk 2]
```diff
@@ -247,7 +248,9 @@ def v602(self):
 		"""An invalid Calendar Quarter was reported. 1) Calendar Quarter must equal 4, and cannot be left blank."""
 		field = "calendar_quarter"
 		edit_name = "v602"
-		fail_df = self.ts_df[self.ts_df.calendar_quarter!=str(4)]
+		fail_df = self.ts_df.copy()
+		fail_df.calendar_quarter = fail_df.calendar_quarter.apply(lambda x: int(x))
```
Comment (Kibrael) [inline python/rules_engine.py]: All the others are string comparisons, why does this one need int?

File: python/rules_engine.py (hunk 1 above)
Comment (MichaelFurey17) [reply python/rules_engine.py]: It has the same logic but the change converts all variables in the check to a string, so the validation uses all similarly-typed variables.

File: python/rules_engine.py (hunk 2 above)
Comment (MichaelFurey17) [reply python/rules_engine.py]: The data map has a configuration for calendar quarter stored as an integer. This changes is to have the validation use all variables with similar types.

**Pull request merged**